Integrate frontend with CI pipeline

navot1234 opened the pull request:
- Add Node.js setup and npm dependency installation
- Configure linting step
- Add build validation and preview testing
- Update README with installation and running instructions
- Clean up unused assets (react.svg, vite.svg, empty directories)

navot1234 committed:
Integrate frontend with CI pipeline

- Add Node.js setup and npm dependency installation
- Configure linting step
- Add build validation and preview testing
- Update README with installation and running instructions
- Clean up unused assets (react.svg, vite.svg, empty directories)

RonKarat1 commented on frontend/src/App.jsx:
```diff
@@ -0,0 +1,94 @@
+
+      <footer className="footer">
+        <div className="container">
+          <p>&copy; 2024 Swappi. All rights reserved.</p>
```
2024 lol

RonKarat1 commented on README.md:
```diff
@@ -12,10 +12,27 @@ Swappi is a web app that lets users seamlessly transfer playlists between Spotif
 
 ## Installation
 
+1. Navigate to the frontend directory:
```
This is not necessary, just add the command where its needed (In the Install dependencies).

RonKarat1 commented on frontend/README.md:
```diff
@@ -0,0 +1,16 @@
+- [@vitejs/plugin-react](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react) uses [Babel](https://babeljs.io/) (or [oxc](https://oxc.rs) when used in [rolldown-vite](https://vite.dev/guide/rolldown)) for Fast Refresh
+- [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react-swc) uses [SWC](https://swc.rs/) for Fast Refresh
+
+## React Compiler
```
I think this is a bit of spam as its not currently supported

navot1234 committed:
Address PR review comments

- Combine cd frontend into install command in README
- Remove React Compiler section from frontend README
- Make copyright year dynamic in App.jsx
